test/default-xrp-client-test.ts

```diff
@@ -461,4 +469,130 @@ describe('Default Xpring Client', function(): void {
       done()
     })
   })
+
+  it('Payment History - successful response', async function(): Promise<void> {
+    // GIVEN a DefaultXRPClient.
+    const xrpClient = new DefaultXRPClient(fakeSucceedingNetworkClient)
+
+    // WHEN the payment history for an address is requested.
+    const paymentHistory = await xrpClient.paymentHistory(testAddress)
+
+    // TODO(amiecorso): should I use the same logic that's in DefaultXRPClient to reconstruct the expected
```

Comment:
I would use the same logic like you're doing. It's test code so you could probably have dropped the switch and just force unwrapped BUT you already have it so I would leave it.

Reply:
I guess this is related to my long comment above about whether this logic belongs in its own method